uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -755,7 +755,8 @@ private static function getBalanceChangesToExpire() {
       WHERE balance_to_expire.expiry_date IS NOT NULL AND
             balance_to_expire.expiry_date < %3 AND
             balance_to_expire.expired_balance_change_id IS NULL AND
-            expired_balance_change.id IS NULL
+            expired_balance_change.id IS NULL AND
+            coalesce(leave_request.type_id, period_entitlement.type_id) IS NOT NULL
```

Comment:
In the description is says

> This is fixed be checking that the Absence Type ID column of the results fetched for the Leave Request corresponding rows is NOT NULL

Why the check for `period_entitlement.id` then?

Reply:
Well, either the leave request type_id field or the period entitlement type_id field must Not be null for a valid result set. Checking for only the Leave request type_id will cause the results for period entitlement to be lost, hence the reason for checking for both.